src/dropin.js

```diff
@@ -246,110 +268,113 @@ Dropin.prototype._initialize = function (callback) {
   }
 
   if (container.innerHTML.trim()) {
-    analytics.sendEvent(this._client, 'configuration-error');
+    analytics.sendEvent(self._client, 'configuration-error');
     callback(new DropinError('options.selector or options.container must reference an empty DOM node.'));
     return;
   }
 
   // Backfill with `en`
-  this._strings = assign({}, translations.en);
-  if (this._merchantConfiguration.locale) {
-    localizedStrings = translations[this._merchantConfiguration.locale] || translations[this._merchantConfiguration.locale.split('_')[0]];
+  self._strings = assign({}, translations.en);
+  if (self._merchantConfiguration.locale) {
+    localizedStrings = translations[self._merchantConfiguration.locale] || translations[self._merchantConfiguration.locale.split('_')[0]];
     // Fill `strings` with `localizedStrings` that may exist
-    this._strings = assign(this._strings, localizedStrings);
+    self._strings = assign(self._strings, localizedStrings);
   }
 
-  if (this._merchantConfiguration.translations) {
-    this._strings = assign(this._strings, this._merchantConfiguration.translations);
+  if (self._merchantConfiguration.translations) {
+    self._strings = assign(self._strings, self._merchantConfiguration.translations);
   }
 
-  localizedHTML = Object.keys(this._strings).reduce(function (result, stringKey) {
-    var stringValue = this._strings[stringKey];
+  localizedHTML = Object.keys(self._strings).reduce(function (result, stringKey) {
+    var stringValue = self._strings[stringKey];
 
     return result.replace(RegExp('{{' + stringKey + '}}', 'g'), stringValue);
-  }.bind(this), mainHTML);
+  }, mainHTML);
 
-  this._dropinWrapper.innerHTML = svgHTML + localizedHTML;
-  container.appendChild(this._dropinWrapper);
+  self._dropinWrapper.innerHTML = svgHTML + localizedHTML;
+  container.appendChild(self._dropinWrapper);
 
-  this._getVaultedPaymentMethods().then(function (paymentMethods) {
-    var paypalRequired;
+  self._getVaultedPaymentMethods().then(function (paymentMethods) {
+    self._model = new DropinModel({
+      client: self._client,
+      componentID: self._componentID,
+      merchantConfiguration: self._merchantConfiguration,
+      paymentMethods: paymentMethods
+    });
 
-    try {
-      this._model = new DropinModel({
-        client: this._client,
-        componentID: this._componentID,
-        merchantConfiguration: this._merchantConfiguration,
-        paymentMethods: paymentMethods
-      });
-    } catch (modelError) {
-      dropinInstance.teardown().then(function () {
-        callback(modelError);
-      });
-      return;
-    }
+    return self._model.initialize();
+  }).then(function () {
+    var paypalRequired;
 
-    this._model.on('cancelInitialization', function (err) {
-      analytics.sendEvent(this._client, 'load-error');
-      this._dropinWrapper.innerHTML = '';
+    self._model.on('cancelInitialization', function (err) {
+      analytics.sendEvent(self._client, 'load-error');
```

Comment:
This is a nitpick but I like reporting analytics just before we exit the function. So in this case, right before the `callback`